climada/hazard/tc_tracks_forecast.py

```diff
@@ -179,76 +184,156 @@ def fetch_bufr_ftp(target_dir=None, remote_dir=None):
 
         return localfiles
 
-    def read_one_bufr_tc(self, file, id_no=None, fcast_rep=None):
+    def read_one_bufr_tc(self, file, id_no=None):
         """ Read a single BUFR TC track file.
 
         Parameters:
             file (str, filelike): Path object, string, or file-like object
             id_no (int): Numerical ID; optional. Else use date + random int.
-            fcast_rep (int): Of the form 1xx000, indicating the delayed
-                replicator containing the forecast values; optional.
-        """
-
-        decoder = pybufrkit.decoder.Decoder()
-
-        if hasattr(file, 'read'):
-            bufr = decoder.process(file.read())
-        elif hasattr(file, 'read_bytes'):
-            bufr = decoder.process(file.read_bytes())
-        elif Path(file).is_file():
-            with Path(file).open('rb') as i:
-                bufr = decoder.process(i.read())
+        """     
+        # Open the bufr file
+        bufr = ec.codes_bufr_new_from_file(file)
+        # we need to instruct ecCodes to expand all the descriptors
+        # i.e. unpack the data values
+        ec.codes_set(bufr, 'unpack', 1)
+        
+        # get the forcast time
+        timestamp_origin = dt.datetime(
+            ec.codes_get(bufr, 'year'), ec.codes_get(bufr, 'month'),
+            ec.codes_get(bufr, 'day'), ec.codes_get(bufr, 'hour'),
+            ec.codes_get(bufr, 'minute'),
+        )
+        timestamp_origin = np.datetime64(timestamp_origin)
+        
+        # number of timestep
+        n_timestep = 0
+        while True:
+            n_timestep = n_timestep + 1
+            try:
+                ec.codes_get_array(bufr, "#%d#timePeriod" % n_timestep)
+            except ec.CodesInternalError:
+                LOGGER.info('Key values not found: end of the time period')
```

Comment:
Please expand this logging info. This is not clear enough.

Reply:
This is a while loop to obtain the number of time steps. I changed to `LOGGER.info('No. of timestep: %d', n_timestep)`

Reply:
This is not a very good way to handle this. You are catching some arbitrary exception and then you do not handle it. I strongly suggest you rewrite this loop such that you do not use an exception as a stoping criterion.